test/e2e/create_test.go

```diff
@@ -246,3 +246,25 @@ func (tc *testContext) waitForWindowsNodes(nodeCount int32, waitForAnnotations,
 
 	return err
 }
+
+// waitForMachineDeletion waits for a certain amount of time for the Windows Machines to be deleted
+func (tc *testContext) waitForMachineDeletion() error {
```

Comment:
This function has overlapping functionality with the [waitForWindowsMachines](https://github.com/openshift/windows-machine-config-operator/blob/1f2bf433fe2e812df22cbc4879d3b0437f8e5788/test/e2e/create_test.go#L110) function

Instead of introducing this new function it would be better to make use of the existing one.

Reply:
my main reasoning to not reuse that function due to two reasons: it has some vsphere platform specific time limits additionally when a machine is being deleted it could quickly change phases and waitForWindowsMachines waits for the machine to be in a particular phase which could lead to potential misses.

Reply:
Thanks for the reasoning, I understand that thought process

The vsphere platform specific time limits were put in place to deal with the root cause of this issue, the machine controller being slow on vsphere. I think it is actually beneficially to have that on the deletion side as well.

> it could quickly change phases 

That is true but calling `waitForWindowsMachines(0)`  will wait until there are 0 Machines, and skip the phase checking. This function is essentially the same as `waitForWindowsNodes` which is being used the same way in the deletion test, specifying that we wait for `0` nodes

Reply:
I think I can make it work for both as you mentioned just not sure if we should go for a time limit of 20 minutes as in case of creation, might seem like an overkill. We could stick to 10 for vsphere and 5 for rest WDYT?

Reply:
I'm not sure that might not be enough. @aravindhp might have more insight.